forc-pkg/src/manifest.rs

```diff
@@ -263,13 +265,20 @@ impl Manifest {
             || self.pkg_dep(STD).is_some()
             || self.dep(STD).is_some()
         {
+            if !include_implicit_std {
+                println_yellow_err("Warning: You specified an STD path but set `implicit-std` to `false`.\
+                 Your custom std path will take precdence and be imported regardless.");
+            }
```

Comment:
I think it should be fine not to warn about this. Eventually we should add some more extensive `Forc.toml` manifest documentation to the Sway book that describes in detail what each of the fields in the manifest mean, similar to [cargo's manifest documentation](https://doc.rust-lang.org/cargo/reference/manifest.html). Under the `implicit_std` field documentation, we can list these conditions under which `std` is not included implicitly.

Reply:
Ups, should have been an else there for those includes... anyway, if warnings aren't needed your alternative solution is much more elegant, implemented it that way.